plugin_ide.ui/src/com/github/rustdt/ide/ui/editor/RustEditorActionContributor.java

```diff
@@ -28,6 +47,88 @@ protected RustOpenDefinitionOperation createOpenDefinitionOperation(ITextEditor
 	
 	@Override
 	protected void registerOtherEditorHandlers() {
+		setupOpenType();

```

Comment:
Thanks for the hint! It helped me a lot understanding how the plugin is set up. As a result, I created the classes `RustIndexManager` and `IndexManager` as siblings to the existing `RustSourceModelManager` and `SourceModelManager`. But I did not include them to the `SourceModelManager` for two reasons:
1. Indexing and parsing/error checking are two different concepts that often behave completely differently and should remain separate in my eyes.
2. The structure of the class is a bit confusing. Especially those nested non-static classes drive me crazy ;-). Would be great if we could improve the readability here.
Besides that, I am wondering how the language agnostic *Open Type* dialog will work in the end. We could just call `LangCore.getIndexManager()` but probably there is a better solution in place.